app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+<?php
+
+namespace App\Http\Controllers\V2\Terrafund;
+
+use App\Http\Controllers\Controller;
+use App\Models\V2\PolygonGeometry;
+use App\Models\V2\Sites\CriteriaSite;
+use App\Models\V2\Sites\SitePolygon;
+use App\Models\V2\WorldCountryGeneralized;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Str;
+use Symfony\Component\Process\Process;
+
+class TerrafundCreateGeometryController extends Controller
+{
+    public function processGeometry(string $uuid)
+    {
+        $geometry = PolygonGeometry::where('uuid', $uuid)
+          ->select(DB::raw('ST_AsGeoJSON(geom) AS geojson'))
+          ->first();
+
+        $geojson = $geometry->geojson;
+
+        if ($geojson) {
+            return response()->json(['geometry' => $geojson], 200);
+        } else {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+    }
+
+    public function storeGeometry(Request $request)
+    {
+        $validatedData = $request->validate([
+          'geometry' => 'required|json',
+        ]);
+        $geometry = json_decode($request->input('geometry'));
+        $geom = DB::raw("ST_GeomFromGeoJSON('" . json_encode($geometry) . "')");
+        $uuid = Str::uuid();
+
+        DB::table('polygon_geometry')->insert([
+          'uuid' => $uuid,
+          'geom' => $geom,
+          'created_at' => now(),
+          'updated_at' => now(),
+        ]);
+
+        return response()->json(['uuid' => $uuid], 200);
+    }
+
+    private function insertSinglePolygon(array $geometry, int $srid)
+    {
+        try {
+            // Convert geometry to GeoJSON string with specified SRID
+            $geojson = json_encode(['type' => 'Feature', 'geometry' => $geometry, 'crs' => ['type' => 'name', 'properties' => ['name' => "EPSG:$srid"]]]);
+
+            // Insert GeoJSON data into the database
+            $geom = DB::raw("ST_GeomFromGeoJSON('$geojson')");
+            $uuid = Str::uuid();
+            $areaSqDegrees = DB::selectOne("SELECT ST_Area(ST_GeomFromGeoJSON('$geojson')) AS area")->area;
+            $latitude = DB::selectOne("SELECT ST_Y(ST_Centroid(ST_GeomFromGeoJSON('$geojson'))) AS latitude")->latitude;
+            $areaSqMeters = $areaSqDegrees * pow(111320 * cos(deg2rad($latitude)), 2);
+
+            $areaHectares = $areaSqMeters / 10000;
+
+            $id = DB::table('polygon_geometry')->insertGetId([
+              'uuid' => $uuid,
+              'geom' => $geom,
+              'created_at' => now(),
+              'updated_at' => now(),
```

Comment:
Same guidance as above on `uuid`, and timestamps.

Reply:
suggestion implemented